### DISCUSSION TITLE
Syncing between instances

### DISCUSSION THREAD

**Discussion by jakubjanousek** [ToolJet/ToolJet#3233] [Q&A]
Hello,
is there a way to sync config/apps between multiple instances? We have multiple clusters with their own databases we would like to self-host tooljet on and since the database structure is the same, we would like to be able to have the same tooljet apps on all clusters.

Is there a way to do this currently with tooljet? It would be preferred if we could have the app versions stored in git, but any other working way would be welcome as well.

Thank you for your answers.

**Comment by akshaysasidrn** [comment]
This is not possible at the moment. Our app doesn’t have a differentiation b/w dev and prod setup. We consider all versions which aren’t released to be up for development and the one released as production. This means dev and prod happen over the same instance. And if users want to segregate development over one instance and production over another, they’d have to export the app definition from the development instance and then import it to production.
But such an approach can bring about certain pain points such as:
- The app versions in exported definition in development instance and then imported at production instance can get out of sync.
- Each new change done in the staging instance would have to be exported and then imported at the production instance as a new app
- Imported apps would need all the creds for data sources in each version to be set each time as they are tied to the specific app

We have planned for introducing git sync, which would sync your app over a git repository and pull in changes when required from a development instance. We shall prioritise it soon.
